src/containers/LocationStorageContainer.tsx

```diff
@@ -0,0 +1,97 @@
+import {Container} from 'unstated'
+import SInfo from 'react-native-sensitive-info'
+
+export interface Location {
+  latitude: number
+  longitude: number
+  when: string // ISO 8601 in UTC (ends in Z)
+}
+
+export interface LocationParams {
+  latitude: number
+  longitude: number
+  when: Date | string
+}
+
+export interface LocationsState {
+  isFetching: boolean
+  locations: Location[]
+  locationsLoaded: boolean
+}
+
+const LOCATIONS_KEY = 'nightingale_locations'
+const SINFO_OPTIONS = {}
+
+/**
+ * An unstated container for managing the secure local storage of
+ * _this user's_ location data
+ */
+export default class LocationStorageContainer extends Container<
+  LocationsState
+> {
+  constructor(_props = {}) {
+    super()
+    this.state = {
+      isFetching: false,
+      locations: [],
+      locationsLoaded: false,
+    }
+  }
+
+  getLocations = async (force = false) => {
+    if (this.state.locationsLoaded && !force) {
+      return this.state.locations
+    }
+
+    await this.setState({
+      isFetching: true,
+    })
+
+    const locationsEncoded = await SInfo.getItem(LOCATIONS_KEY, SINFO_OPTIONS)
+
+    await this.setState({
+      isFetching: false,
+      locations: JSON.parse(locationsEncoded || "[]"),
+      locationsLoaded: true,
+    })
+    return this.state.locations
+  }
+
+  pruneLocations = async () => {
+    if (!this.state.locationsLoaded) {
+      await this.getLocations()
+    }
+
+    // TODO: prune the pings that happened more than 2 weeks ago
+
+    return this.saveLocations()
+  }
+
+  recordLocation = async (locationParams: LocationParams) => {
+    if (!this.state.locationsLoaded) {
+      await this.getLocations()
+    }
+    // normalize the date
```

Comment:
@jrotolo @DejaTrudeaux I was thinking we could add friendly location info to the `Location` type (but not the `LocationParams` type) and merge that into the object here before we merge it into the locations list. Open to other ideas, though